azavea/pfb-network-connectivity
# Feature/kak/job status filter speed#400

## Overview

Reduce the number and complexity of queries and operations performed for commonly used filter to get last completed or last run analysis job for a neighborhood. Closes #400, which should also close #373.


### Notes

Denormalizes job status, start time (from when import began), and final run time. Reduced queries for latest analysis jobs from 25 to four, two of which are for neighborhoods. Neighborhood queries could possibly be also reduced to one, but added `select_related` to filter query had no effect, likely due to the related queryset defined for the custom `PrimaryKeyReferenceRelatedField` on the serializer.


## Testing Instructions

To enable database query debugging, add the following clause under 'loggers' in the settings:
```python
'django.db.backends': {
    'handlers': ['console'],
    'level': 'DEBUG',
},
```

 * Run migrations
 * Should have added fields set after migrations (i.e. `last_job` set on Neighborhood objects)
 * Query http://localhost:9200/api/analysis_jobs/?format=json&latest=True&limit=8&ordering=-overall_score&status=COMPLETE
 * Query results should output same results as previously


Connects #373 Closes #400
